upgrade-2.x.sh

```diff
@@ -218,7 +218,7 @@ function upgrade_supervisor() {
                     curl -s "${API_ENDPOINT}/v2/device($DEVICEID)?apikey=$APIKEY" -X PATCH -H 'Content-Type: application/json;charset=UTF-8' --data-binary "{\"supervisor_release\": \"$UPDATER_SUPERVISOR_ID\"}" > /dev/null 2>&1
                     log "Running supervisor updater..."
                     progress 90 "Running supervisor update"
-                    update-resin-supervisor
+                    update-resin-supervisor || log WARN "Supervisor couldn't be updated, continuing anyways"
```

Comment:
Shouldn't we allow fails here only for the rpi case on the known broken versions?

Reply:
@agherzan Yeah, was thinking about it. IMHO the update failure is generally not a breaking issue, and since it recovers by itself, it is probably not helpful to have an "OS update failed" which agents have to fix up, instead of the device self-recovering.

Reply:
Fair.